tests/test_sensor.py

```diff
@@ -80,6 +82,49 @@ def status_func(value):
     assert sensor.status == Sensor.Status.NOMINAL
 
 
+def test_sensor_value_setter(diverse_sensors):
```

Comment:
Split this into two sets of tests: one for success and one for failure. Then use `pytest.mark.parametrize` to reduce boilerplate. For the success tests, you'll have parameters for the sensor name, set value, and value to read back. For failure tests, the sensor name and the value to set.

I'd also suggest ditching the `diverse_sensors` fixture and just creating a sensor on the fly inside the test (replacing the sensor name parameter with a sensor type). That way you can additionally test creating a sensor of that type with the value as a default (for both success and failure cases).